packages/vite/src/node/plugins/asset.ts

```diff
@@ -182,6 +185,10 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
     },
 
     generateBundle(_, bundle) {
+      duplicateAssets.forEach((asset) => {
+        bundle[asset.name!] = asset
+      })
```

Comment:
I ran this PR locally, and noticed that this indeed fixes the missing dedupe assets when generating a manifest. However according to the manifest example you gave:

```json
{
  "resources/images/client1/icon.png": {
    "file": "assets/icon.636a0396.png",
    "src": "resources/images/client1/icon.png"
  },
  "resources/images/client2/icon.png": {
    "file": "assets/icon.71e56d11.png",
    "src": "resources/images/client2/icon.png"
  },
  "resources/images/client3/icon.png": {
    "file": "assets/icon.636a0396.png",
    "src": "resources/images/client3/icon.png"
  },
}
```

The third entry's `file` is the same as the first entry's only coincidentally if you have a duplicated of `/client1/icon.png` and `/client2/icon.png`. If we have `/client/icon.png` and `/client/icon2.png` instead (where both is identical. The third entry `file` would be `"assets/icon2.636a0396.png"` instead.

Should we have the third entry reference the deduped asset instead?

(Reproduced via duplicating `nested/asset2.png` in the `assets` playground, import it in JS, and the manifest showed this pattern)

---

Also expanding on `icon.png` and `icon2.png`, for non-SSR setups, the client build still always refers to the deduped asset (`icon.png`). So the extra generated asset isn't being referenced anywhere, which may be wasteful.

I think a nicer solution is to keep a list of duplicated assets, and then add them back through the `manifest.ts` plugin so only the manifest is affected for now.

Reply:
See: https://github.com/vitejs/vite/pull/9928#issuecomment-1234991179